shaders/MicrobeBackground.gdshader

```diff
@@ -1,64 +1,77 @@
 shader_type spatial;
 render_mode unshaded;
 
-// this is 1 by default to keep the dark BGs looking nice
+// Default light level for dark backgrounds
 uniform float lightLevel = 1.0f;
 
 uniform vec2 repeats = vec2(1.0f, 1.0f);
-uniform sampler2D layer0 : source_color;
-uniform sampler2D layer1 : source_color;
-uniform sampler2D layer2 : source_color;
-uniform sampler2D layer3 : source_color;
+uniform sampler2D layer0 : source_color; 
+uniform sampler2D layer1 : source_color; 
+uniform sampler2D layer2 : source_color; 
+uniform sampler2D layer3 : source_color; 
 
-const vec2 speed0 = vec2(3300.0f);
-const vec2 speed1 = vec2(2550.0f);
-const vec2 speed2 = vec2(1800.0f);
-const vec2 speed3 = vec2(1050.0f);
+// the noise textures, distortion amount, and distortion speed for the distortion shader
+uniform sampler2D noise_tex_1; 
+uniform sampler2D noise_tex_2; 
+uniform float distortion_factor; 
+uniform float distortion_speed; 
 
-varying vec2 UV3;
-varying vec2 UV4;
+const vec2 speed0 = vec2(3300.0f); 
+const vec2 speed1 = vec2(2550.0f); 
+const vec2 speed2 = vec2(1800.0f); 
+const vec2 speed3 = vec2(1050.0f); 
 
-vec3 LightInfluence(float amount)
-{
-    if (amount > 0.5f)
-    {
-        // Day
+varying vec2 UV3; 
+varying vec2 UV4; 
+
+vec3 LightInfluence(float amount) { 
+    if (amount > 0.5f) {
         return mix(vec3(0.75f, 0.5f, 0.5f), vec3(1.0f, 1.0f, 1.0f), 2.0f * amount - 1.0f);
-    }
-    else if (amount > 0.25f)
-    {
-        // Dawn and Dusk
+    } else if (amount > 0.25f) {
         return mix(vec3(0.25f, 0.25f, 0.25f), vec3(0.75f, 0.5f, 0.5f), 4.0f * amount - 1.0f);
-    }
-    else
-    {
-        // Night
+    } else {
         return mix(vec3(0.052f, 0.05f, 0.17f), vec3(0.25f, 0.25f, 0.25f), 4.0f * amount);
     }
 }
 
-void vertex(){
+void vertex() { 
     vec2 offset = (repeats - 1.0f) / 2.0f;
     vec2 worldPos = (INV_VIEW_MATRIX * vec4(0.0f, 0.0f, 0.0f, 1.0f)).xz;
 
-    UV = (UV + worldPos / (speed0 * repeats)) * repeats - offset;
-    UV2 = (0.12f + UV + worldPos / (speed1));
-    UV3 = (0.512f + UV + worldPos / (speed2));
-    UV4 = (0.05f + UV + worldPos / (speed3));
+    UV = (UV + worldPos / (speed0 * repeats)) * repeats - offset; 
```

Comment:
And a lot of these spaces at the end of lines cause sections to be marked as changed. Any good text editor should have a feature to delete all trailing whitespace so these should be easy to get rid of.